lib/models/application.coffee

```diff
@@ -28,6 +31,15 @@ getApplicationModel = (deps, opts) ->
 
 	exports = {}
 
+	# Internal method for name/id disambiguation
+	# Note that this throws an exception for missing names, but not missing ids
+	getId = (nameOrId, callback) ->
+		(if isId(nameOrId)
```

Comment:
This might be nicer as
```coffee
Promise.try ->
	if isId(nameOrId)
		return nameOrId
	else
		exports.get(nameOrId).get('id')
.asCallback(callback)